### DISCUSSION TITLE
How many keyword is too many

### DISCUSSION THREAD

**Discussion by trimbilrepo** [tweepy/tweepy#1631] [Questions] [answered]
I just wondering is there any limit that we can put in on the _track_ parameters using Tweepy stream function?

```python
myStream.filter(track=keywords, is_async=True)
```

I tried to add 15 and more keywords in the same time. Sure, it return tweet, but not all keyword give me results. I don't know if i reach keyword limit (if there is any) or Twitter just doesn't send me.
I'm not sure if the total return amount of the tweet for every keyword is in range of 1% of the tweets.

What I know is that Twitter API only return 1% of the their twitter data. However, I don't know is it 1% for every keyword or something else.

Let assume there are 1.000.000 tweet and consist of words: 

1. python 500.000 tweets
2. tweepy 250.000 tweets
3. github 150.000 tweets
4. stackoverflow 100.000 tweets

add all words to the stream filter:
```python
myStream.filter(track=["python","tweepy","github","stackoveflow"], is_async=True)
``` 
Could we get 1% of every keyword? or it will be 1% of total all keyword? or if first keyword already reach 1% or rate limit, it will not returning other keyword?

**Comment by Harmon758** [comment, marked as answer]
The limit for keywords for the POST statuses/filter endpoint that `Stream.filter` uses is 400.
https://developer.twitter.com/en/docs/twitter-api/filtered-stream-overview

https://twittercommunity.com/t/post-statuses-filter-limits/23179 is an old post, but likely still applies.
If so, it would imply that the stream would return Tweets matching any and all of the keywords up to 1% before limiting you.
It might be worth asking this question on the [Twitter Developers Forums](https://twittercommunity.com/) though, as it's not specific to Tweepy.

**Reply by trimbilrepo** [reply]
Thank you, it give me more understanding about the limit.
